ckanext/s3filestore/uploader.py

```diff
@@ -53,9 +58,14 @@ def __init__(self):
         self.region = config.get('ckanext.s3filestore.region_name')
         self.signature = config.get('ckanext.s3filestore.signature_version')
         self.download_proxy = config.get('ckanext.s3filestore.download_proxy', None)
+        self.signed_url_expiry = int(config.get('ckanext.s3filestore.signed_url_expiry', '3600'))
+        self.signed_url_cache_window = int(config.get('ckanext.s3filestore.signed_url_cache_window', '1800'))
         self.acl = config.get('ckanext.s3filestore.acl', 'public-read')
         self.session = None
 
+    def is_url_caching_enabled(self):
+        return self.signed_url_cache_window and self.signed_url_cache_window > 0 and self.signed_url_expiry and self.signed_url_expiry > 0
```

Comment:
do we need to do this complex if statement multiple times. could we store this in the class and only do it if its null?

Reply:
Since this is called at least twice and as it does not change until a reboot of python, it could be stored in self.cache_enabled or do the calc on __init__  after the config is loaded.

Do you know if the BaseS3Uploader is init once and called multiple times?